src/tool/cppwinrt/strings/base_collections_base.h

```diff
@@ -103,17 +103,35 @@ namespace winrt
 
             uint32_t GetMany(array_view<T> values)
             {
-                uint32_t const actual = (std::min)(static_cast<uint32_t>(std::distance(m_current, m_end)), values.size());
-                m_owner->copy_n(m_current, actual, values.begin());
-                std::advance(m_current, actual);
-                return actual;
+                if constexpr (std::is_same_v<decltype(typename std::iterator_traits<iterator_type>::iterator_category()), std::random_access_iterator_tag>)
+                {
+                    uint32_t const actual = (std::min)(static_cast<uint32_t>(m_end - m_current), values.size());
+                    m_owner->copy_n(m_current, actual, values.begin());
+                    m_current += actual;
+                    return actual;
+                }
+                else
+                {
+                    auto output = values.begin();
+
+                    while (output < values.end() && m_current != m_end)
+                    {
+                        *output = Current();
+                        ++output;
+                        ++m_current;
+                    }
+
+                    return static_cast<uint32_t>(output - values.begin());
```

Comment:
Directly subtracting iterators is an operation that is only valid for `random_access_iterator_tag`, which this 'else' clause is explicitly not supposed to use. Do you have test coverage exercising this branch?

Reply:
Yes, this is heavily tested. In fact, I just added even more rigorous tests for this branch. #499

Reply:
I think the confusion here is that the subtraction is not on the input but on the output. ;)

Reply:
Oh yeah, that's totally safe to do on the output. Thanks for pointing that out. 👍